osu.Game.Rulesets.Osu/UI/Cursor/CursorTrail.cs

```diff
@@ -285,9 +285,11 @@ protected override void Draw(IRenderer renderer)
                     if (time - part.Time >= 1)
                         continue;
 
+                    Vector2 screenSpacePos = Source.ToScreenSpace(part.Position);
```

Comment:
This should not be done in the DrawNode.

Reply:
For those that don't know why, can you provide an explanation of what is wrong and why?

Reply:
Simply because this is accessing a member which is on the update thread. For instance, this could cause validations which is a recursive query up the scene graph with who knows what effects - the important part is that it's a cross thread access.

The only time you should access "Source" is inside `ApplyState`.

Reply:
Thanks, assuming you're gonna push a fix?

Reply:
Yeah I'm looking into it.

Reply:
Sorry for my bad implementation, I should have been aware of that

Reply:
All good, it's why we have reviews :)